### ISSUE TITLE
www.habitica.com --- links to profile and notifications not recognized

### ISSUE THREAD

**Issue by dexic79** [philc/vimium#3144]
Hi! :-)

I am very thankful for your piece of software, because it makes my life very easy during the day. Almost no mouse needed, which is a relief for me, who has had some trouble with his hand due to a lot of mouse usage.

I just wanted to let you know, that two of the main links of www.habitica.com don't work, when you are logged in. This is a wonderful to do web software and helps me to get through my days a lot!

Thank you in advance!

Cheers!

**Comment by kenny-evitt** [comment]
@dexic79 There's a bunch of 'links' in Habitica that Vimium can't 'click' because of the specific HTML elements Habitica uses.

Interestingly, as-of just now, both the 'profile' and 'notifications' items (at the top right of the page) *are* both 'clickable', at least for me.

Some links that aren't but that I wish were are the "Due" link for 'Dailies' and "Scheduled" link for 'To-Dos'. That's because both are `div` elements which Vimium, rightfully, doesn't consider to be 'link' elements. More frustratingly, at one point, those links *were* 'clickable' with Vimium, but whatever change was made was soon after undone, effectively.

In general, problems like this aren't the fault of, nor easily solved by, Vimium. They're a consequence of 'poorly structured' websites where 'link elements' are not constructed of a suitable HTML elements whereby Vimium can reasonably infer that they should be 'clickable'.

Because of all of the above, I think this issue should be closed.

**Comment by gdh1995** [comment]
I agree with what the kenny-evitt say, and it's somehow hard to fix.

But here's an advertisement: I make a customized version of Vimium - https://github.com/gdh1995/vimium-c - and it hooks almost all click listeners from pages' JavaScript code and record their target nodes, so Vimium C is able to list much more clickable items. As a side effect, it will hint some butons twice or some unexpected container nodes. Then you may have a try.

Once Vimium also hooks click listeners, but its implementation had some bugs and was given up on 2018/04/01 in commit 5745bb344959a61297fc16e00bb6de0a2a8744d9.
